### ISSUE TITLE
[feature] Time cartesian axes

### ISSUE THREAD

**Issue by vramosx** [vramosx/flutter_web_chartjs#4]
[feature request]
Support for Time Cartesian Axis:

https://www.chartjs.org/docs/latest/axes/cartesian/time.html

**Comment by JaviBonilla** [comment]
Right now, it is only supported the [category](https://www.chartjs.org/docs/latest/axes/cartesian/category.html) type in the x axis, such this:

```
ChartJS(
    id: 'line-category',
    config: ChartConfig(
        type: ChartType.line, 
        data: ChartData(
            labels: ['Jan', 'Feb', 'Mar'],
            datasets: [
                ChartDataset(
                    data: [10, 20, 30],
                ),
            ],
        ),
    ),
)
```
But, it could be also supported the [time](https://www.chartjs.org/docs/latest/axes/cartesian/time.html) axis. This requires [Moments.js](https://momentjs.com/) or [any other date library supported](https://www.chartjs.org/docs/master/axes/cartesian/time.html#date-adapters) by Chart.js.

```
ChartJS(
    id: 'line-time',
    config: ChartConfig(
        type: ChartType.line,
        options: ChartOptions(
            scales: ChartScales(
                xAxes: [ChartAxis(type: ChartAxisType.time)],
            ),
        ),  
        data: ChartData(
            datasets: [
                ChartDataset(
                    data: [
                        {'t': '2020-01-01', 'y': 10},
                        {'t': '2020-02-01', 'y': 20},
                        {'t': '2020-03-01', 'y': 30},
                    ],
                ),
            ],
        ),
    ),
)
```

An the [linear](https://www.chartjs.org/docs/latest/axes/cartesian/linear.html) axis.

```
ChartJS(
    id: 'line-linear',
    config: ChartConfig(
        type: ChartType.line,
        options: ChartOptions(
            scales: ChartScales(
                xAxes: [ChartAxis(type: ChartAxisType.linear)],
            ),
        ),  
        data: ChartData(
            datasets: [
                ChartDataset(
                    data: [
                        {'x': 1, 'y': 10},
                        {'x': 2, 'y': 20},
                        {'x': 3, 'y': 30},
                    ],
                ),
            ],
        ),
    ),
)
```
I can work on adding some of these features, if it is ok for you. It involves changing the *ChartDataset* **data** from *List< double >* to *List< dynamic >* and adding some enums and  variables for additional options.

**Comment by vramosx** [comment]
Hi @JaviBonilla,

Yes, you can add features and I will appreciate your help! :)

I just have a concern, with this feature, moment.js will be required even if it's not used(like in category type axis)? If it can be optional it's better. 

Cheers

**Comment by JaviBonilla** [comment]
@vramosx, no problem with that, moment.js is optional, everything (besides the time axis) will keep working as usual.

**Comment by vramosx** [comment]
Thanks, @JaviBonilla!
Available in v0.2.5--alpha.
